src/context/WalletProvider/KeepKey/setupKeepKeySdk.ts

```diff
@@ -0,0 +1,20 @@
+import { KeepKeySdk } from '@keepkey/keepkey-sdk'
+
+export const setupKeepKeySDK = async () => {
+  let serviceKey = window.localStorage.getItem('@app/serviceKey')
+  let config: any = {
+    apiKey: serviceKey,
+    pairingInfo: {
+      name: 'ShapeShift',
+      imageUrl: 'https://assets.coincap.io/assets/icons/fox@2x.png',
+      basePath: 'http://localhost:1646/spec/swagger.json',
+      url: 'https://app.shapeshift.com',
+    },
+  }
+  let sdk = await KeepKeySdk.create(config)
+
+  if (!serviceKey) {
+    window.localStorage.setItem('@app/serviceKey', config.apiKey)
```

Comment:
how does this ever get set? does the `KeepKeySdk.create()` call set this in localstorage?

Reply:
the sdk mutates the config object after pairing. if the apiKey is not set it is issued by the wallet. the sdk does not touch localstorage, dapps must store their own api key for the next session.

Reply:
please add a comment to this effect, that the `create()` call mutates the input args, as this otherwise looks like a bug.

interesting design choice, and something i would not recommend. i'd suggest the below or similar.

```
const { sdk, apiKey } = await KeepKeySdk.create(config)
```

Reply:
updated comments